pkgs/by-name/ha/harlequin/package.nix

```
@@ -0,0 +1,64 @@
+{
+  lib,
+  python3,
+  fetchFromGitHub,
+  withPostgresAdapter ? true,
+  withBigQueryAdapter ? true,
+}:
+python3.pkgs.buildPythonApplication rec {
+  pname = "harlequin";
+  version = "1.24.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "tconbeer";
+    repo = "harlequin";
+    rev = "ref/tags/v${version}";
+    hash = "sha256-exgpdixYc0hpB0GGw20Apg1rlw3lhStJaIX0UHPu10A=";
+  };
+
+  build-system = with python3.pkgs; [
+    poetry-core
+  ];
+
+  dependencies =
+    with python3.pkgs;
+    [
+      textual
+      textual-fastdatatable
+      textual-textarea
+      click
+      rich-click
+      duckdb
+      sqlfmt
+      platformdirs
+      importlib-metadata
+      tomlkit
+      questionary
+      numpy
+      packaging
+    ]
+    ++ lib.optionals withPostgresAdapter [ harlequin-postgres ]
+    ++ lib.optionals withBigQueryAdapter [ harlequin-bigquery ];
+
+  pythonRelaxDeps = [
+    "textual"
+  ];
+
+  pythonImportsCheck = [
+    "harlequin"
+    "harlequin_duckdb"
+    "harlequin_sqlite"
+    "harlequin_vscode"
+  ];
+
+  meta = {
```

```suggestion
  ];
  
  passthru = {
    updateScript = nix-update-script { };
    tests.version = testers.testVersion { package = harlequin; };
  };

  meta = {
```
Don't forget to add `harlequin` to the derivation inputs.

---

Thank you. I added it, but surprisingly, I'm encountering issues.

```
took 4s ➜  nix-build -A harlequin.tests.version                     
this derivation will be built:
  /nix/store/slcc8v51d4cwli39sssv1riryqhfxl3m-harlequin-1.24.0-test-version.drv
building '/nix/store/slcc8v51d4cwli39sssv1riryqhfxl3m-harlequin-1.24.0-test-version.drv'...
harlequin --version returned a non-zero exit code.
Harlequin could not load the installed plug-in named google.cloud.

No module named 'google.cloud'
Traceback (most recent call last):
  File "{{storeDir}}/bin/.harlequin-wrapped", line 9, in <module>
    sys.exit(harlequin())
             ^^^^^^^^^^^
  File "{{storeDir}}/lib/python3.12/site-packages/harlequin/cli.py", line 413, in harlequin
    cli = build_cli()
          ^^^^^^^^^^^
  File "{{storeDir}}/lib/python3.12/site-packages/harlequin/cli.py", line 153, in build_cli
    adapters = load_adapter_plugins()
               ^^^^^^^^^^^^^^^^^^^^^^
  File "{{storeDir}}/lib/python3.12/site-packages/harlequin/plugins.py", line 17, in load_adapter_plugins
    return _load_plugins(group="harlequin.adapter")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "{{storeDir}}/lib/python3.12/site-packages/harlequin/plugins.py", line 62, in _load_plugins
    return plugins
           ^^^^^^^
UnboundLocalError: cannot access local variable 'plugins' where it is not associated with a value
```

This is strange as if I just `nix-build -A harlequin` and try to use it with `harlequin -a bigquery` it works perfectly fine. :thinking: 

What am I missing?

---

This test is simply running `harlequin --version`.
Is this command supposed to exist ? Currently, it is failing with the message you are seeing.

---

Yes the command works fine if I do `nix run .\#harlequin -- --version`

```
harlequin, version 1.24.0

Installed Adapters:
  - duckdb, version 1.24.0
  - sqlite, version 1.24.0
  - postgres, version 0.4.0
  - bigquery, version 1.0.2
```